content/sql/concepts/window-functions/terms/lag/lag.md

```diff
@@ -29,33 +32,106 @@ OVER ( [ partition_by ] order_by )
+    sales;
 ```
 
-The output is a table that features a new column `previous_age`, which holds the values from the previous records. The first record is null because a default was not specified and the previous row would be out of range.
+The output is a table with a new column `previous_order_date`, which contains previous entries of `order_date` in the table when ordered by `order_date`. The offset is set to 1 by default, and NULL is displayed when the offset is out of range.
 
 Output
 
-| first_name | last_name | age | previous_age |
-| ---------- | --------- | --- | ------------ |
-| kyle       | xy        | 60  | null         |
-| jenna      | black     | 35  | 60           |
-| chris      | smith     | 30  | 35           |
-| dave       | james     | 19  | 30           |
+| product_id | order_date | sold_price | previous_order_date |
+| ---------- | ---------- | ---------- | ------------------- |
+|          1 | 2024-01-01 |        100 | NULL                |
+|          2 | 2024-01-01 |         50 | 2024-01-01          |
+|          1 | 2024-02-01 |        120 | 2024-01-01          |
+|          2 | 2024-02-01 |         60 | 2024-02-01          |
+|          2 | 2024-02-02 |         60 | 2024-02-01          |
+|          1 | 2024-03-01 |         80 | 2024-02-02          |
+
+Example 2 :
+
+The below example uses `LAG()` with offset of 1, partitioned by `product_id` and Order by `order_date`.
+
+```sql
+SELECT
+    product_id,
+    order_date,
+    sold_price,
+    LAG(sold_price,1) OVER (
+      PARTITION BY product_id 
+      ORDER BY order_date
+      ) AS previously_sold_price
+FROM
+    sales;
+```
+
+The output is a table that features a new column `previously_sold_price`, which holds the sold price of the product on the previous sold date. The first record of every `product_id` is NULL because a default was not specified as the previous date is out of range for the first sale of the product.
+
+Output
+
+| product_id | order_date | sold_price | previously_sold_price |
+| ---------- | ---------- | ---------- | --------------------- |
+|          1 | 2024-01-01 |        100 |                  NULL |
+|          1 | 2024-02-01 |        120 |                   100 |
+|          1 | 2024-03-01 |         80 |                   120 |
+|          2 | 2024-01-01 |         50 |                  NULL |
+|          2 | 2024-02-01 |         60 |                    50 |
+|          2 | 2024-02-02 |         60 |                    60 |
+
+Example 3:
+
+The below example uses `LAG()` with offset of 1 and with default value "First Sale".
+
+```sql
+SELECT
+    product_id,
+    order_date,
+    sold_price,
+    LAG(product_id,1,"First Sale") OVER (
+      ORDER BY order_date
+      ) AS previous_sold_product
+FROM
+    sales;
+  
+```
+
+This outputs column `previous_sold_product`, which contains the previous sold product for each sale ordered by `order_date`. As there is no previous sale for the first entry, the default is set to "First Sale".
```

Comment:
We can simplify the explanation by specifying that the Product ID is returned in the `previous_sold_product` column and adding more detail to explain when the default value is applied, as follows -

```suggestion
This query outputs a new column, `previous_sold_product`, which contains the product ID from the previous sale for each row ordered by order_date. If there is no previous sale (for the first entry), the default value "First Sale" is used.
```